grr/server/grr_response_server/gui/ui/components/code_editor/code_editor.ts

```diff
@@ -0,0 +1,84 @@
+import {
+    ChangeDetectionStrategy,
+    Component,
+    ViewChild,
+    ElementRef,
+    AfterViewInit,
+    ViewEncapsulation,
+    Input,
+    forwardRef,
+} from '@angular/core';
+import { NG_VALUE_ACCESSOR, ControlValueAccessor } from '@angular/forms';
+
+import * as CodeMirror from 'codemirror';
+// importing sql-hint is needed for SQL syntax highlighting
+import 'codemirror/addon/hint/sql-hint.js';
+// importing show-hint is needed for the autocomplete pop-up
+import 'codemirror/addon/hint/show-hint.js';
+
+type OnChangeFn = (textValue: string) => void;
+
+/** Displays a code editor. */
+@Component({
+  selector: 'code-editor',
+  templateUrl: './code_editor.ng.html',
+  styleUrls: ['./code_editor.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  encapsulation: ViewEncapsulation.None,
+  providers: [
+    {
+      provide: NG_VALUE_ACCESSOR,
+      useExisting: forwardRef(() => CodeEditor),
+      multi: true,
+    },
+  ],
+})
+export class CodeEditor implements AfterViewInit, ControlValueAccessor {
+  @ViewChild('editorTarget')
+  private readonly editorTarget!: ElementRef;
+  private editor?: CodeMirror.Editor;
+
+  private latestOverwrite = '';
+  @Input()
+  set editorValue(newValue: string) {
+    this.latestOverwrite = newValue;
+    this.editor?.setValue(newValue);
```

Comment:
Do I understand it correctly that editor might be not set when this function is called? Let's add a comment why that might be the case.